networkx/utils/backends.py

```diff
@@ -1160,7 +1160,9 @@ def _convert_graph(
             )
             if cache:
                 warning_message = (
-                    f"Using cached graph for {backend_name!r} backend in "
+                    "Note: conversions to backend graphs are saved to cache "
+                    "(`G.__networkx_cache__` on the original graph) by default."
```

Comment:
I know this isn't the first time we've documented `__networkx_cache__` but do we want to advertise our "private" members?  Would it be better to just add a well-named property for formally inspecting things like the state of the cache now and document that?  That would give us the freedom to possibly change things related to caching later without breaking docs/users.

OTOH, keeping it a dunder makes it obvious to a user that it's not related to "their graph" (IOW, it's NX bookkeeping and not user data), so I could go either way.  I think I'm leaning toward making it a property though.

Reply:
I don't understand this suggestion, or maybe I just don't agree with it. Dunders aren't private. `x.foo` is public, `x._foo` is an implementation detail, `x.__foo` is private (and mangled), and `x.__foo__` is a dunder. Dunder methods are different. They are good for protocols, even if the protocol is simply "here there be caching". I think it is helpful to indicate that the cache exists on the object, which is why I advertise it.

I think introducing e.g. `G.cache` is out of scope for this PR. Similarly, helper functions for working with the cache is another thing I think we've discussed (not today, but other days) could make sense, and when we have them we can update this message.

In other words, I don't think this feedback is actionable for this PR, but is something we can be aware of and discuss (and perhaps track) elsewhere.